src/FlashtestationRegistry.sol

```diff
@@ -4,30 +4,38 @@ pragma solidity 0.8.28;
 import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
 import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {ReentrancyGuardTransient} from "@openzeppelin/contracts/utils/ReentrancyGuardTransient.sol";
+import {EIP712Upgradeable} from "@openzeppelin/contracts-upgradeable/utils/cryptography/EIP712Upgradeable.sol";
 import {IAttestation} from "./interfaces/IAttestation.sol";
+import {IFlashtestationRegistry} from "./interfaces/IFlashtestationRegistry.sol";
 import {QuoteParser, WorkloadId} from "./utils/QuoteParser.sol";
 import {TD10ReportBody} from "automata-dcap-attestation/contracts/types/V4Structs.sol";
 import {TD_REPORT10_LENGTH, HEADER_LENGTH} from "automata-dcap-attestation/contracts/types/Constants.sol";
 
-// TEE identity and status tracking
-struct RegisteredTEE {
-    WorkloadId workloadId; // The workloadID of the TEE device
-    bytes rawQuote; // The raw quote from the TEE device, which is stored to allow for future quote quote invalidation
-    bool isValid; // true upon first registration, and false after a quote invalidation
-    bytes publicKey; // The 64-byte uncompressed public key of TEE-controlled address, used to encrypt messages to the TEE
-}
-
 /**
  * @title FlashtestationRegistry
  * @dev A contract for managing trusted execution environment (TEE) identities and configurations
  * using Automata's Intel DCAP attestation
  */
-contract FlashtestationRegistry is Initializable, UUPSUpgradeable, OwnableUpgradeable {
+contract FlashtestationRegistry is
+    IFlashtestationRegistry,
+    Initializable,
+    UUPSUpgradeable,
+    OwnableUpgradeable,
+    EIP712Upgradeable,
+    ReentrancyGuardTransient
+{
+    using ECDSA for bytes32;
+
     // Constants
 
     // Maximum size for byte arrays to prevent DoS attacks
     uint256 public constant MAX_BYTES_SIZE = 20 * 1024; // 20KB limit
 
+    // EIP-712 Constants
+    bytes32 public constant REGISTER_TYPEHASH = keccak256("RegisterTEEService(bytes rawQuote,uint256 nonce)");
```

Comment:
Consider following a similar format as UniswapX and other repositories where we create a library specifically for each 712 hashed struct which contains the typeHash as well as hashing functions. Even if you only anticipate having only one Struct I feel it's nicer to separate out all of the 712 related logic from core logic. For reference: https://github.com/Uniswap/calibur/blob/main/src/libraries/SignedBatchedCallLib.sol

Reply:
That makes sense, but it won't be as clean to implement it liked `SignedBatchedCallLib` because of how we're using OZ's `EIP712Upgradeable`, which stores some state (specifically the `EIP712Storage` struct) which can get changed as versions upgrade.

If I remove upgradeability in this contract I will implement your suggestion, but I think for now, with the code already written and using OZ, I'd rather keep it as it is.